models/stdp_triplet_connection.h

```diff
@@ -185,7 +187,7 @@ class STDPTripletConnection : public Connection< targetidentifierT >
   facilitate_( double_t w, double_t kplus, double_t ky )
   {
     double_t new_w = w + kplus * ( Aplus_ + Aplus_triplet_ * ky );
-    return new_w < Wmax_ ? new_w : Wmax_;
+    return new_w < std::abs( Wmax_ ) ? new_w : std::abs( Wmax_ );
```

Comment:
Looking at this code again after a comment from @sanjayankur31 which is strangely missing from Github, I find the code very difficult to read. The problem is that on l. 254f you call call `facilitate_()` with the absolute value of `w`, something that is not clear from the code here. Basically, you mix handling of signs in `send()` and `facilitate_()`. I would suggest to do all signe handling inside `facilitate_()`, so that the code in `send()` would again become
```
weight_ = facilitate_( weight_, Kplus_ * std::exp( minus_dt / tau_plus_ ), ky );
```
That way, all sign handling will be inside `facilitate_()`, which we then could code like this
```
const double_t new_w_abs = std::abs( w ) + kplus * ( Aplus_ + Aplus_triplet_ * ky );
return copysign( std::min( new_w_abs, std::abs( Wmax_ ) ), Wmax_ );
```
I suppose the same applies to `depress_()`.